### PR TITLE
Add NuGet strong-name/publishing metadata to .NET packages

### PR DESCRIPTION
**Pull request by mpiroc**
Add all relevant metadata needed to sign and publish `Amazon.JSII.JsonModel`, `Amazon.JSII.Generator`, and `Amazon.JSII.Runtime` to [nuget.org](https://www.nuget.org/).

This metadata includes the path to a `key.snk`. This file will be retrieved by a script in a separate pull request that I'll send out shortly. If `key.snk` is not present, it will be ignored and .NET assemblies will not be strong-named.

### CODE REVIEW

File: packages/jsii-dotnet-generator/Directory.Build.props.t.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
-const path = require('path');
-const version = require('./package.json').version.replace(/\+.+$/, ''); // omit "+build" postfix;
+const version = require('./package.json').version.replace(/\+.+$/, ''); // omit "+build" suffix
```
Comment (eladb) [inline packages/jsii-dotnet-generator/Directory.Build.props.t.js]: Why do we need to publish the code generator to NuGet? Isn't it just bundled as part of `jsii-pacmak`? I don't see a use case for someone to use it directly...

File: packages/jsii-dotnet-jsonmodel/NuGet.Metadata.props.t.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const package = require('./package.json');
+
+process.stdout.write(`<Project>
+  <PropertyGroup>
+    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
+    <PackageOutputPath>..\\..\\bin\\$(Configuration)\\NuGet\\</PackageOutputPath>
```
Comment (eladb) [inline packages/jsii-dotnet-jsonmodel/NuGet.Metadata.props.t.js]: Will the backslash be a problem on UNIX systems?

Comment (mpiroc) [reply packages/jsii-dotnet-jsonmodel/NuGet.Metadata.props.t.js]: Nope, tested on WSL and `superchain`.

File: packages/jsii-dotnet-generator/Directory.Build.props.t.js (hunk 1 above)
Comment (mpiroc) [reply packages/jsii-dotnet-generator/Directory.Build.props.t.js]: It isn't necessary, I just included it for completeness. Will remove.

**Pull request merged**